homeassistant/components/fyta/sensor.py

```diff
@@ -0,0 +1,180 @@
+    ),
+    FytaSensorEntityDescription(
+        key="temperature_status",
+        translation_key="temperature_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="light_status",
+        translation_key="light_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="moisture_status",
+        translation_key="moisture_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="salinity_status",
+        translation_key="salinity_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="light",
+        translation_key="light",
+        native_unit_of_measurement="mol/d",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="moisture",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.MOISTURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="salinity",
+        translation_key="salinity",
+        native_unit_of_measurement="mS/cm",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="ph",
+        device_class=SensorDeviceClass.PH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    FytaSensorEntityDescription(
+        key="plant_number",
+        translation_key="plant_number",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the FYTA binary sensors."""
+    coordinator: FytaCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    plant_entities: list[CoordinatorEntity] = []
+    plant_entities = [
+        FytaCoordinatorSensor(coordinator, entry, sensor)
+        for sensor in SENSORS
+        if sensor.key in coordinator.data or sensor.key == "plant_number"
```

Comment:
Instead of the or, you could just add it to the list separately (so outside of a for loop)

Reply:
This was the easiest solution I could think of. But I'm curious to see how you would implement it.

Reply:
`plant_entities.append(FytaCoordinatorSensor(coordinator, entry, PLANT_NUMBER_SENSOR))`

You then make sure that the plant number sensor is its own constant instead of in the list

Reply:
Ok... Seems more complicated to me. But let me know if this should be required.

Reply:
It's really not that complicated, it's just adding another entry to a list :)

It will improve performance since it doesn't have to compare the key every time